override font defaults

quacklabs opened the pull request:
Modified custom Init and made text resize based on height of the component and device type. font is overridden and set based on detected device type. results in bigger fonts for bigger screens and initial font setting for smaller screens.

quacklabs committed:
override font defaults

Modified custom Init and made text resize based on height of the component and device type. font is overridden and set based on detected device type. results in bigger fonts for bigger screens and initial font setting for smaller screens.

quacklabs committed:
override font defaults

Modified custom Init and made text resize based on height of the component and device type. font is overridden and set based on detected device type. results in bigger fonts for bigger screens and initial font setting for smaller screens.

quacklabs committed:
override font defaults

Modified custom Init and made text resize based on height of the component and device type. font is overridden and set based on detected device type. results in bigger fonts for bigger screens and initial font setting for smaller screens.

quacklabs committed:
override font defaults

Modified custom Init and made text resize based on height of the component and device type. font is overridden and set based on detected device type. results in bigger fonts for bigger screens and initial font setting for smaller screens.

quacklabs committed:
Fine tuned font size constants

Refactored the font resize parameters that was causing fonts to be too small on smaller screens

quacklabs committed:
Fine tuned font size constants

Refactored the font resize parameters that was causing fonts to be too small on smaller screens

quacklabs committed:
Fine tuned font size constants

Refactored the font resize parameters that was causing fonts to be too small on smaller screens

quacklabs committed:
Fine tuned font size constants

Refactored the font resize parameters that was causing fonts to be too small on smaller screens

quacklabs committed:
Removed all print statements and comments

Removed debug print statements that were not needed in production.

quacklabs committed:
Removed all print statements and comments

Removed debug print statements that were not needed in production.

quacklabs committed:
Removed all print statements and comments

Removed debug print statements that were not needed in production.

quacklabs committed:
Removed all print statements and comments

Removed debug print statements that were not needed in production.

itswisdomagain committed:
Create .gitignore

itswisdomagain commented on AutoResizeFont/DynamicLabel.swift:
```diff
@@ -10,38 +10,60 @@ import UIKit
 @IBDesignable
 final class DynamicLabel: UILabel {
+    
+    //allow font to accept size from interface builder and scale up from there.
```
- Comments ending with a period should begin with capital letter.
- There should be a space before the first letter of a comment.

quacklabs replied:
Fixed in commit #5cf1b4a

itswisdomagain commented on AutoResizeFont/DynamicLabel.swift:
```diff
@@ -10,38 +10,60 @@ import UIKit
-    var fontSizePts: CGFloat {
-        didSet {
-            calculateFontSize()
+    var pointSize = UIFont() {
```
`pointSize` is not a size but a `UIFont`

quacklabs replied:
I am overriding "font". using self.font.withSize returns a UIFont and not a size value. basically, I am just replicating the font, applying the changes and assigning it back to the default "font" instance. if you do have a better approach, I would love to learn.

quacklabs replied:
the font.pointSize value cannot be reassigned a new value as it is a get-only property. I did try getting just the size and applying but it is not allowed by default.

itswisdomagain replied:
I actually meant that the variable would need renaming. It says "size" but it's a "font"

quacklabs replied:
okay. fixed in commit #000054f

itswisdomagain commented on AutoResizeFont/DynamicLabel.swift:
```diff
@@ -10,38 +10,60 @@ import UIKit
-        fontSizePts = 14
-        super.init(frame: frame);
+        super.init(frame: frame)
+        relativeHeight = CGFloat(frame.height * relativeFontConstant)
```
`frame.height` refers to the height of the label in question. Not sure that's a good reference point. A dev could assign any height to a label. Even if a dev doesn't do so, and the height is not fixed, then the height of the label would be dependent on the font size. This `frame.height` value should be the same for all devices as long as the font size is the same (which it is until the font is auto-resized, and the font's auto-resizing relies on this value).

itswisdomagain replied:
Do you think the algorithm used [here](https://stackoverflow.com/questions/53495725/how-to-ios-item-and-font-sizes-scaling-as-screen-size/53496831) will work? That code uses static font size increments for different device types, adding +2, +2, +3, +4 to the base font size depending on the device.

quacklabs replied:
the font relies on its label height, which is not much different from the font size measured in pts. I think it is a good point to reference because it will resize the font based on set autoLayout constraints. I tried the +2 +3 approach. the font still remains small  with hardly any noticeable difference on bigger devices

itswisdomagain replied:
I see. Will test your PR shortly and see how it works.

itswisdomagain commented on AutoResizeFont/DynamicLabel.swift:
```diff
@@ -10,38 +10,60 @@ import UIKit
     
-    // todo calculate fonts for iPad and iPod
+    //calculate fonts for iPad, iPhone and iPod
     func calculateFontSize() {
```
`calculateFontSize` is only called when the view is initialised. If the font size is changed subsequently, this calculation is not triggered.

quacklabs replied:
I am only calculating and applying the font properties on load of the view. are you referring to changes that may be made on the device due to accessibility settings?. Please clarify.

itswisdomagain replied:
No, changes that could be made while designing on IB or changes that could be effected from code after the app is already running.

quacklabs replied:
fixed in commit #618c4f7

itswisdomagain commented on AutoResizeFont/ViewController.swift:
```diff
@@ -9,7 +9,9 @@
 class ViewController: UIViewController {
-
+    
+    @IBOutlet var label1 : DynamicLabel!
```
This outlet is not necessary.

quacklabs replied:
I know, I only added it as a test. there are two labels but I only created one outlet for testing purposes.

quacklabs replied:
it does not have to be in the production code, will remove it as directed

itswisdomagain reviewed (changes requested):
I see some progress here but some modifications are in place. See the comments below.
Also, remove the `.DS_Store` and all `*/xcuserdata/*` files from source control. I've added a `.gitignore` to master to prevent unwanted files getting into source control subsequently.

quacklabs committed:
Fixed comments

Fixed comments ending with period; starting them with a capital letter and added spaces before the first letter of each comment

quacklabs committed:
Fixed comments

Fixed comments ending with period; starting them with a capital letter and added spaces before the first letter of each comment

quacklabs committed:
Fixed comments

Fixed comments ending with period; starting them with a capital letter and added spaces before the first letter of each comment

quacklabs committed:
Fixed comments

Fixed comments ending with period; starting them with a capital letter and added spaces before the first letter of each comment

quacklabs committed:
renamed resize font var

renamed the resized font from pointSize to resizedFont

quacklabs committed:
renamed resize font var

renamed the resized font from pointSize to resizedFont

quacklabs committed:
renamed resize font var

renamed the resized font from pointSize to resizedFont

quacklabs committed:
renamed resize font var

renamed the resized font from pointSize to resizedFont

quacklabs committed:
font size is inspectable in IB

made font size inspectable in Interface builder, removed duplicate resizeFont var. font sets and renders automatically while using interface builder now.

quacklabs committed:
font size is inspectable in IB

made font size inspectable in Interface builder, removed duplicate resizeFont var. font sets and renders automatically while using interface builder now.

quacklabs committed:
font size is inspectable in IB

made font size inspectable in Interface builder, removed duplicate resizeFont var. font sets and renders automatically while using interface builder now.

quacklabs committed:
font size is inspectable in IB

made font size inspectable in Interface builder, removed duplicate resizeFont var. font sets and renders automatically while using interface builder now.

quacklabs committed:
update userinterface

quacklabs committed:
update userinterface

quacklabs committed:
updated gitignore

quacklabs committed:
updated gitignore

quacklabs committed:
updated gitignore

quacklabs committed:
updated gitignore

quacklabs committed:
resolved gitignore

quacklabs committed:
resolved gitignore

quacklabs committed:
resolved gitignore

quacklabs committed:
resolved gitignore

itswisdomagain commented on AutoResizeFont/DynamicLabel.swift:
```diff
@@ -10,38 +10,68 @@ import UIKit
-    var fontSizePts: CGFloat {
-        didSet {
+    var fontSize: CGFloat {
+        get{
```
Why not using `didSet`?

quacklabs replied:
didSet does not give me direct access to newValue on update. if i am to use the value immediately it is set, I need to use set.

itswisdomagain replied:
Or you can use `willSet`

itswisdomagain commented on .gitignore:
```diff
@@ -0,0 +1,68 @@
+!default.perspectivev3
+xcuserdata/
+# .DS_Store
+# */xcuserdata/*
```
Why are these commented (lines 19 and 20). We want them ignored from the repo.
I still see these files in your PR, we can't merge them into master.

itswisdomagain commented on .gitignore:
```diff
@@ -0,0 +1,68 @@
+# fastlane/report.xml
+# fastlane/Preview.html
+# fastlane/screenshots/**/*.png
+# fastlane/test_output
```
Why are these commented?
Also, add an empty line at the end of this file.

itswisdomagain commented on AutoResizeFont/Base.lproj/Main.storyboard:
```diff
@@ -18,22 +18,30 @@
+                                    <userDefinedRuntimeAttribute type="number" keyPath="fontSize">
+                                        <real key="value" value="34"/>
+                                    </userDefinedRuntimeAttribute>
+                                    <userDefinedRuntimeAttribute type="number" keyPath="fontSizePts">
```
I don't think we should be having this anymore. Can it be removed?

quacklabs replied:
These are automatically added by xcode, altering them may prevent the project from building. i encountered something similar while working on [https://github.com/quacklabs/SwiftValidator](url) and it prevented the project from passing build tests for hours.

itswisdomagain replied:
That's an interesting observation.

itswisdomagain commented on AutoResizeFont/ViewController.swift:
```diff
@@ -9,7 +9,9 @@
 class ViewController: UIViewController {
-
+    
+    @IBOutlet var label1 : DynamicLabel!
```
We still have this outlet.
I'd prefer if this file is unchanged, don't see a need to change anything here.

quacklabs replied:
fixed in commit #b25317a

itswisdomagain commented on AutoResizeFont/DynamicLabel.swift:
```diff
@@ -10,38 +10,68 @@ import UIKit
     }
     
+    // Deduced by multiplying the height of the Label by a constant.
+    var relativeHeight : CGFloat!
```
Should be `relativeHeight: CGFloat`
- Don't think the `!` is necessary.
- Also, no spacing before the colon.

quacklabs replied:
the ! is neccesary or the compiler will give us errors in the init() call. 
spacing fixed in commit #b25317a

itswisdomagain replied:
I see. I understand the compiler error.

itswisdomagain commented on AutoResizeFont/DynamicLabel.swift:
```diff
@@ -10,38 +10,68 @@ import UIKit
+    
+    // Relative constant to multiply our initial element height.
+    // This lets us double the size on bigger devices
+    let relativeFontConstant:CGFloat = 0.1
```
Single space after colon.

quacklabs replied:
fixed in commit #b25317a

quacklabs committed:
fixed changes suggested in review

quacklabs committed:
fixed changes suggested in review

quacklabs committed:
fixed changes suggested in review

quacklabs committed:
fixed changes suggested in review

quacklabs committed:
deleted my gitignore

quacklabs committed:
deleted my gitignore

quacklabs committed:
Merge branch 'AutoresizeText' of https://github.com/quacklabs/custom-label-dynamic-device-font-size into AutoresizeText

quacklabs committed:
Merge branch 'AutoresizeText' of https://github.com/quacklabs/custom-label-dynamic-device-font-size into AutoresizeText

quacklabs committed:
updated gitignore

quacklabs committed:
Update UserInterfaceState.xcuserstate
